src/main/java/com/achdev/onlinebookstoreapp/dto/user/UserRegistrationRequestDto.java

```diff
@@ -23,8 +23,9 @@ public class UserRegistrationRequestDto {
             message = "Password must be between 8 and 35 characters")
     @Pattern(
             regexp = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&*+=]).{8,35}$",
-            message = "Password must contain at least one digit, one lowercase letter, "
-                      + "one uppercase letter, and one special character"
+            message = "Password must be between 8 and 35 characters long and "
+                      + "contain at least one digit, one lowercase letter, "
+                      + "one uppercase letter, and one special character."
```

Comment:
You should add the registration method by giving each user who registers a default USER role. Passing roles in the body of the registration request is not secure, as this is an open endpoint and any user can set any role for themselves.

Reply:
Ok